nixos/modules/services/misc/nix-daemon.nix

```diff
@@ -192,50 +192,136 @@ in
+              '';
+            };
+            maxJobs = mkOption {
+              type = types.int;
+              default = 1;
+              description = ''
+                The number of concurrent jobs the build machine supports. The
+                build machine will enforce its own limits, but this allows hydra
+                to schedule better since there is no work-stealing between build
+                machines.
+              '';
+            };
+            speedFactor = mkOption {
+              type = types.int;
+              default = 1;
+              description = ''
+                The relative speed of this builder. This is an arbitrary integer
+                that indicates the speed of this builder, relative to other
+                builders. Higher is faster.
+              '';
+            };
+            mandatoryFeatures = mkOption {
+              type = types.listOf (types.enum [ "big-parallel" "kvm" "benchmark" "nixos-test" ]);
+              default = [];
+              example = [ "big-parallel" ];
+              description = ''
+                A list of features mandatory for this builder. The builder will
+                be ignored for derivations that don't require all features in
+                this list. All mandatory features are automatically included in
+                <varname>supportedFeatures</varname>.
+                Note that there is no fixed set of features. To avoid
+                misconfiguration, this option only accepts a set of typical
+                values.
+                Please use the <varname>mandatoryExtraFeatures</varname> option
+                for mandatory custom features.
+              '';
+            };
+            mandatoryExtraFeatures = mkOption {
+              type = types.listOf types.str;
+              default = [];
+              description = ''
+                A list of additional mandatory features.
+                See <varname>mandatoryFeatures</varname> for more information.
+              '';
+            };
+            supportedFeatures = mkOption {
+              type = types.listOf (types.enum [ "big-parallel" "kvm" "benchmark" "nixos-test" ]);
+              default = [];
+              example = [ "kvm" ];
+              description = ''
+                A list of features supported by this builder. The builder will
+                be ignored for derivations that require features not in this
+                list.
+                Note that there is no fixed set of features. To avoid
+                misconfiguration, this option only accepts a set of typical
+                values.
+                Please use the <varname>mandatoryExtraFeatures</varname> option
+                for mandatory custom features.
+              '';
+            };
+            supportedExtraFeatures = mkOption {
+              type = types.listOf types.str;
+              default = [];
+              example = [ "big-mistake-this-ones-impure" ];
```

Comment:
Is this option impure, or you're saying there could be a feature named impure? This implies the option itself is a big mistake and impure :).

Reply:
I don't know. [You tell me](https://github.com/NixOS/nixpkgs/pull/47636#discussion_r221797702) :)

Reply:
This option is not impure :) there could be a feature named `impure`, though. I was trying to nudge you towards rewording the documentation through a joking hypothetical.